compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -1194,11 +1195,22 @@ fn trait_object_ty<'tcx>(tcx: TyCtxt<'tcx>, poly_trait_ref: ty::PolyTraitRef<'tc
                 .filter(|item| item.kind == ty::AssocKind::Type)
                 .map(move |assoc_ty| {
                     super_poly_trait_ref.map_bound(|super_trait_ref| {
-                        let alias_ty = ty::AliasTy::new(tcx, assoc_ty.def_id, super_trait_ref.args);
+                        // While vtable-safe projections can't have const or type parameters that
+                        // aren't part of the trait_ref, they *can* have additional regions. Pad out
+                        // the projection with erased regions.
+                        let missing_args =
+                            tcx.generics_of(assoc_ty.def_id).count() - super_trait_ref.args.len();
+                        let extra_regions =
+                            iter::repeat(tcx.lifetimes.re_erased.into()).take(missing_args);
+                        let alias_args = tcx.mk_args_from_iter(
+                            super_trait_ref.args.into_iter().chain(extra_regions),
+                        );
```

Comment:
but in any case, this code could be replaced with sth. like

```suggestion
                        let missing_args =
                            tcx.generics_of(assoc_ty.def_id).count() - super_trait_ref.args.len();
                        let extra_regions =
                            iter::repeat(tcx.lifetimes.re_erased.into()).take(missing_args);
                        let alias_args = tcx.mk_args_from_iter(
                            .into_iter().chain(extra_regions),
                        );
                        let alias_args = super_trait_ref.args.extend_to(tcx, assoc_ty.def_id, |..| tcx.lifetimes.re_erased.into());
```

However, I wouldn't advice you to use `'erased` unconditionally for all arguments, I would the very least panic/bug if the `param.kind` is not a region.

Reply:
Note that is just a drive-by review, I don't have enough context is your approach makes sense in general.

Reply:
`extend_to` is what I would recommend here, but we shouldn't need to do this at all.

Reply:
Good to know for the future :)